SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1674,6 +1676,44 @@ USpatialActorChannel* ActorSystem::TryRestoreActorChannelForStablyNamedActor(AAc
 	return Channel;
 }
 
+void ActorSystem::InvokeRepNotifies()
+{
+	for (FObjectRepNotifies& ObjectRepNotifies : RepNotifiesToSend)
+	{
+		UObject* Object = ObjectRepNotifies.Object.Get();
+		if (!IsValid(Object))
+		{
+			continue;
+		}
+		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
+		if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			continue;
+		}
+		USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
+		if (!IsValid(Channel))
+		{
+			continue;
+		}
```

Comment:
What happened with the warning you had here before and the test triggering it?
Did you discover why that was? I think it's a potentially important edge-case.

Reply:
I can imagine channel warning here being hit if an actor is deleted, and a subobject on it with repnotifies isn't - so could have been that.

That is a good question though, what is the expected behaviour of subobjects whos parent actor has been deleted?

Reply:
I believe subobjects get destroyed if the parent actor gets destroyed.... maybe

Reply:
seems to only unregister them - see UWorld::DestroyActor:807. I think unregistering is different from marking pendingkill? https://docs.unrealengine.com/4.26/en-US/ProgrammingAndScripting/ProgrammingWithCPP/UnrealArchitecture/Actors/Components/#registeringcomponents and https://docs.unrealengine.com/4.26/en-US/BlueprintAPI/Components/DestroyComponent/ give me that impression

(so the IsValid check above our code here would pass, though I guess we could also put an IsRegistered check and then logs, but would have to cast to actorcomponent, feels messy etc)

Reply:
though I am surprised... it feels like something that is just a memory leak by leaving them hanging around in memory - so it does feel like I've missed something :/

Reply:
```
// Clean up the actor's components.
	ThisActor->UnregisterAllComponents();

	// Mark the actor and its direct components as pending kill.
	ThisActor->MarkPendingKill();
	ThisActor->MarkPackageDirty();
	ThisActor->MarkComponentsAsPendingKill();
```
?